specs/commandline/policy.md

```diff
@@ -0,0 +1,162 @@
+
+- The Policy named "wabbit-networks-images" is for verifying images signed by Wabbit Networks and stored in two repositories `registry.acme-rockets.io/software/net-monitor` and `registry.acme-rockets.io/software/net-logger`.
+
+- Policy named "unsigned-image" is for skipping the verification on unsigned images stored in repository `registry.acme-rockets.io/software/unsigned/net-utils`.
+- Policy "allow-expired-images" is for logging instead of failing expired images stored in repository `"registry.acme-rockets.io/software/legacy/metrics`.
+- Policy "global-policy-for-all-other-images" is for verifying any other images that signed by the ACME Rockets.
+  
+```jsonc
+{
+    "version": "1.0",
+    "trustPolicies": [
+        {
+            "name": "wabbit-networks-images",
+            "registryScopes": [ 
+              "registry.acme-rockets.io/software/net-monitor",
+              "registry.acme-rockets.io/software/net-logger" 
+            ],
+            "signatureVerification": {
+                "level": "strict"
+            },
+            "trustStores": [ 
+              "ca:wabbit-networks",
+            ],
+            "trustedIdentities": [
+                "x509.subject: C=US, ST=WA, L=Seattle, O=wabbit-networks.io, OU=Security Tools"
+            ]
+        },
+        {
+            "name": "unsigned-image",
+            "registryScopes": [ "registry.acme-rockets.io/software/unsigned/net-utils" ],
+            "signatureVerification": {
+              "level" : "skip" 
+            }
+        },
+        {
+            "name": "allow-expired-images",
+            "registryScopes": [ "registry.acme-rockets.io/software/legacy/metrics" ],
+            "signatureVerification": {
+              "level" : "strict",
+              "override" : {
+                "expiry" : "log"
+              }
+            },
+            "trustStores": ["ca:acme-rockets"],
+            "trustedIdentities": ["*"]
+        },
+        {
+            "name": "global-policy-for-all-other-images",
+            "registryScopes": [ "*" ],       
+            "signatureVerification": {                                
+                "level": "strict"
+            },
+            "trustStores": [ 
+              "ca:acme-rockets"
+            ],                  
+            "trustedIdentities": [                                    
+                "x509.subject: C=US, ST=WA, L=Seattle, O=acme-rockets.io, CN=SecureBuilder"
+            ]
+        }
+    ]
+}
+```
+
+## Outline
+
+### notation policy command
+
+```text
+Manage trust policies for signature verification.
+
+Usage:
+  notation policy [command]
+
+Available Commands:
+  export    export trust policies to a JSON file
```

Comment:
I'm trying to understand the use case here. Are we trying to export the trustpolicy.json to another JSON file?

Reply:
Yes. The purpose for `import` and `export` commands is users don't need to know where the trust policies are stored for different OS, and what the name of the trust policy configuration file should be.

Reply:
Also, user can generate and test trust policy locally and export it, then import it in there CI/CD or production environment.

Reply:
I see. Then what if there isn't an existing trustpolicy.json under the user's config dir, for example, when the user just downloaded the notation binary. How would the user get started in this case? IMO, a `notation policy init` command is required here.

Reply:
This is the first iteration focusing on `import` and `export`. To get started, users can copy the template from the spec into a file. We can improve it on the next iteration.

Reply:
> 

I'm not sure, "users can copy the template from the spec into a file" sounds not a very good user experience because in this way, the user still needs to know where they should put this file OR still call `notation policy import` as an extra command. IMO, we should include this `notation policy init` method in phase-1, once this command is called, a default trust policy should be generated and put at the correct place for the user automatically.

Reply:
The problem is that when we generate the default policy, users still need to export it, update it, and then import it. It will take more time to get an aligned UX, so the template will be put to next iteration.

Reply:
> The problem is that when we generate the default policy, users still need to export it, update it, and then import it. It will take more time to get an aligned UX, so the template will be put to next iteration.

Then the question becomes can we generate a good enough default policy that satisfy most beginners/new users? For example, a default trust policy with all levels `strict` as global registry scope. Because for a 0-experience starter, going to the specs to find a trust policy template is very hard. A workaround is that we explicitly print out a template for users to use when notation detects the trustpolicy.json is missing.
Btw, correct me if I'm wrong, phase-1 targets RC.4, and phase-2 targets later than RC.4 right? Will we have both phases ready before the Kubecon?

Reply:
Let's find a time to discuss the default policy to see whether we can make it by RC-4.